airflow/models/taskinstance.py

```diff
@@ -376,7 +379,8 @@ def log_url(self):
             "execution_date={iso}"
             "&task_id={task_id}"
             "&dag_id={dag_id}"
-        ).format(iso=iso, task_id=self.task_id, dag_id=self.dag_id)
+            "&try_number={try_number}"
+        ).format(iso=iso, task_id=self.task_id, dag_id=self.dag_id, try_number=self.try_number)
```

Comment:
Personally I don't think this change here is necessary. When we provide the log page url, by default we're going to give the latest attempt. Having this change or not doesn't change anything.

All other parts look good to me, other than another minor suggestion for the test.

Reply:
We might want to send email notification for every fail/retry, this link will lead user always to exact execution log (not to the latest execution like it happening right now)